lib/attrsets.nix

```
@@ -542,6 +542,25 @@ rec {
     attrs:
     map (name: f name attrs.${name}) (attrNames attrs);
 
+  /* Deconstruct an attrset to a list of `nameValuePair`s as expected by
+     `builtins.listToAttrs`. Each element of the resulting list consists of a string-valued
+     attribute name specifying the name of the attribute, and an attribute value specifying
+     its value.
+     Note: While this is almost the inverse of `builtins.listToAttrs`, the builtin might
+     already change the order of the list items and thus cannot be restored when converting
+     back. Additionally, `builtins.listToAttrs` prunes list items with duplicate names
+     and only includes the first occurence in its result.
+
+     Example:
+       attrsToList {foo=1; bar="asdf";}
+       => [ { name = "bar"; value = "asdf"; } { name = "foo"; value = 1; } ]
+
+     Type:
+       attrsToList :: AttrSet -> [ { name:: String; value :: Any; }]
+
+  */
```

```suggestion
  /*
    Deconstruct an attrset to a list of name-value pairs as expected by [`builtins.listToAttrs`](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-listToAttrs).
    Each element of the resulting list is an attribute set with these attributes:
    - `name` (string): The name of the attribute
    - `value` (any): The value of the attribute
    
    The following is always true:
    ```nix
    builtins.listToAttrs (attrsToList attrs) == attrs
    ```
    
    :::{.warning}
    The opposite is not always true however:
    ```nix
    attrsToList (builtins.listToAttrs list) != list
    ```
    
    This is because the `listToAttrs` removes duplicate names and doesn't preserve the order of the list.
    :::

    Example:
      attrsToList { foo = 1; bar = "asdf"; }
      => [ { name = "bar"; value = "asdf"; } { name = "foo"; value = 1; } ]

    Type:
      attrsToList :: AttrSet -> [ { name :: String; value :: Any; }]

  */
```

---

Thanks for the better docs. I still have to get adjusted to Myst-flavoured markdown.